front/js/product.js

```diff
@@ -3,7 +3,7 @@ function displaySelectedProduct() {
   const whatProductUrl = window.location.href;
   const productUrl = new URL(whatProductUrl);
   const search_params = new URLSearchParams(productUrl.search);
-  function verifyIdThenDisplayProduct() {
+  const verifyId = function verifyIdThenDisplayProduct() {
```

Comment:
you should not declare a function inside another function. put it above or bellow and call it in you code